scripts/requirements.sh

```diff
@@ -0,0 +1,365 @@
+parse-mappings golang/github.com/vmware-tanzu/velero /usr/local/bin/velero "https://github.com/vmware-tanzu/velero/releases/download/v\${version}/velero-v\${version}-linux-amd64.tar.gz" "velero-v\${version}-linux-amd64/velero"
+parse-mappings golang/github.com/mikefarah/yq/v4 /usr/local/bin/yq4 "https://github.com/mikefarah/yq/releases/download/v\${version}/yq_linux_amd64"
+
+# <file>
+parse-requirements() {
+  local file="${1}"
+
+  if ! [[ -f "${file}" ]]; then
+    echo "failure, not a file"
+    return 1
+  fi
+
+  local -a rows
+  readarray rows <"${1}"
+
+  for row in "${rows[@]}"; do
+    parse-package "${row}"
+  done
+}
+
+# <purl>
+parse-package() {
+  local purl="${1}" remainder component
+
+  purl="${purl##+([[:space:]])}"
+  purl="${purl%%+([[:space:]])}"
+
+  remainder="${purl}"
+
+  if [[ "${remainder}" =~ "#" ]]; then
+    component="${purl##*"#"}"
+    component="${component##"/"}"
+    # TODO: Split, discard dot and empty segments, decode, and rejoin
+    subpaths["${purl}"]="${component%%"/"}"
+    remainder="${purl%"#"*}"
+  fi
+
+  if [[ "${remainder}" =~ "?" ]]; then
+    # TODO: Manage key value pairs
+    qualifiers["${purl}"]="${remainder##*"?"}"
+    remainder="${remainder%"?"*}"
+  fi
+
+  component="${remainder%%":"*}"
+  if [[ "${component}" != "pkg" ]]; then
+    echo "invalid purl: invalid scheme"
+    return
+  fi
+  remainder="${remainder#*":"}"
+
+  remainder="${remainder##"/"}"
+  remainder="${remainder%%"/"}"
+
+  component="${remainder%%"/"*}"
+  if [[ -z "${component}" ]]; then
+    echo "invalid purl: missing type"
+    return
+  fi
+  types["${purl}"]="${component}"
+  remainder="${remainder#*"/"}"
+
+  if [[ "${remainder}" =~ "@" ]]; then
+    # TODO: Decode
+    versions["${purl}"]="${remainder##*"@"}"
+    remainder="${remainder%"@"*}"
+  fi
+
+  component="${remainder##*"/"}"
+  if [[ -z "${component}" ]]; then
+    echo "invalid purl: missing name"
+    return
+  fi
+  # TODO: Decode and normalise
+  names["${purl}"]="${component}"
+  remainder="${remainder%"/"*}"
+
+  if [[ "${remainder}" =~ "/" ]]; then
+    # TODO: Split, discard empty segments, decode, normalise, and rejoin
+    namespaces["${purl}"]="${remainder}"
+  fi
+
+  packages+=("${purl}")
+
+  # echo "${purl} - pkg: ${types["${purl}"]} / ${namespaces["${purl}"]:-} / ${names["${purl}"]} @ ${versions["${purl}"]:-} ? ${qualifiers["${purl}"]:-} # ${subpaths["${purl}"]:-}"
+}
+
+# <packages...>
+install-from-apt() {
+  apt-get update
+  apt-get upgrade -y
+  apt-get install -y "${@}"
+  apt-get clean
+  rm -rf /var/lib/apt/lists/*
+}
+
+# <target> <source> <checksum-bin>
+install-from-bin() {
+  local target="${1}" source="${2}" checksum_bin="${4:-}" checksum
+
+  echo "installing ${target}"
+
+  if [[ -e "${target}" ]] && [[ -n "${checksum_bin}" ]]; then
+    checksum="$(sha256sum "${target}")"
+    if [[ "${checksum_bin}" == "${checksum%%" "*}" ]]; then
```

Comment:
```suggestion
    if sha256sum --check - <<< "${checksum_bin}  ${target}" >/dev/null 2>/dev/null; then
```

Could save a tiny bit of string wrangling, but should otherwise be equivalent.